apps/projects/routes.py

```diff
@@ -505,242 +552,117 @@ def put(self, project_id):
-                storage_id:
-                  type: string
-                  example: 2019/5/fa5079a38e0a4197864aa2ccb07f3bea.mp4
-                metadata:
-                  type: object
-                  example: {}
-                mime_type:
-                  type: string
-                  example: video/mp4
-                create_time:
-                  type: string
-                  example: 2019-05-01T09:00:00+00:00
-                original_filename:
-                  type: string
-                  example: video.mp4
-                request_address:
-                  type: string
-                  example: 127.0.0.1
-                version:
-                  type: integer
-                  example: 2
-                parent:
-                  type: object
-                  parameters:
-                    _id:
-                      type: object
-                      parameters:
-                        $oid:
-                          type: string
-                          example: 5ccbc4104dfd9b8fa153d60e
                 processing:
                   type: boolean
-                  example: False
-                thumbnails:
-                  type: object
-                  example: {}
-                _id:
-                  type: string
-                  example: 5cbd5acfe24f6045607e51aa
+                  example: True
         """
-        check_user_agent()
-        schema = check_request_schema_validity(request.get_json(), self.SCHEMA_EDIT)
-
-        doc = find_one_or_404(project_id)
-
-        filename, ext = os.path.splitext(doc['filename'])
-        if doc.get('version') >= 2:
-            raise BadRequest("Only POST action for original video version 1")
-        version = doc.get('version', 1) + 1
-        new_file_name = f'{filename}_v{version}{ext}'
-        new_doc = {
-            'filename': new_file_name,
-            'storage_id': doc.get('storage_id'),
-            'metadata': None,
-            'request_address': get_request_address(request.headers.environ),
-            'version': version,
-            'processing': True,
-            'mime_type': doc.get('mime_type'),
-            'parent': {
-                '_id': doc.get('_id'),
-            },
-            'thumbnails': {},
-            'preview_thumbnail': doc.get('preview_thumbnail')
-        }
-        app.mongo.db.projects.insert_one(new_doc)
-        new_doc['predict_url'] = get_url_for_media(new_doc.get('_id'), 'video')
-        task_edit_video.delay(json_util.dumps(new_doc), schema)
-        save_activity_log("POST PROJECT", doc['_id'], doc['storage_id'], schema)
-        return json_response(new_doc)
+
+        if self._project['processing']['video']:
+            return json_response({'processing': True}, status=409)
+
+        if self._project['version'] == 1:
+            raise BadRequest(f"Video with version 1 is not editable, use duplicated project instead.")
```

Comment:
imo it would be good to have one way of sending errors to clients, probably using json plus some user readable message, but here we have json for processing error and text message for version one

Reply:
agree